### PR TITLE
OpenAPI specification

### PR DESCRIPTION
**Pull request by majsa1**
The OpenAPI specification for the routes that are now in use. We could publish the specification after review, in that case the server location needs to be adjusted.

### CODE REVIEW

File: documentation/openAPI/wasted-openAPI.yaml [hunk 1]
```diff
@@ -0,0 +1,625 @@
+openapi: 3.0.3
+info:
+  title: Wasted API - OpenAPI 3.0
+  description: API specification document of the Wasted API
+  version: 1.0.0
+servers:
+  - url: http://localhost:8080/api
```
Comment (NoNamer777) [inline documentation/openAPI/wasted-openAPI.yaml:7]:
```suggestion
  - url: http://localhost:8080/
    description: Local development back-end application
  - url: https://server.wasted.nl.eu.org/
    description: Back-end application (self-hosted)
  - url: https://wasted-back-end-app.azurewebsites.net
    description: Back-end application (hosted via Azure)
```

**Review by NoNamer777 — CHANGES_REQUESTED** [review]
Left some comments.

In general: check the paths again. You now have a base server address of `http://localhost:8080/api` which is not correct. The base server address should be `http://localhost:8080/`. So besides the `/auth/*` paths, all the other paths needs to be prefixed with `/api`

File: documentation/openAPI/wasted-openAPI.yaml [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+  - bearerAuth: [] 
+
+paths:
+  /auth/register:
+    post:  
+      tags:
+        - Auth
+      summary: Registers a user.
+      servers:
+        - url: http://localhost:8080
```
[the remark this answers is not in the document]
Comment (majsa1) [reply documentation/openAPI/wasted-openAPI.yaml:18]: this was added as this is the only route which doesn't have 'api' in the path

[the remark this answers is not in the document]
Comment (majsa1) [reply documentation/openAPI/wasted-openAPI.yaml:18]: never mind, didn't see your comment below yet :-)

**Review by NoNamer777 — DISMISSED** [review]

**Review by NoNamer777 — APPROVED** [review]

**Pull request merged**